apps/oracles/eth-rpc/src/lib.rs

```diff
@@ -0,0 +1,242 @@
+    pub result: Option<String>,
+}
+
+fn convert(x: [u8; 32]) -> ([u8; 16], [u8; 16]) {
+    let arr: [[u8; 16]; 2] = unsafe { std::mem::transmute(x) };
+    (arr[0], arr[1])
+}
+
+impl ResponseEthCall {
+    pub fn result_as_u256(&self) -> Result<U256> {
+        match &self.result {
+            Some(v) => Ok(U256::from_str(v)?),
+            None => {
+                let message = if let Some(err) = &self.error {
+                    err.message.clone()
+                } else {
+                    "Missing error".to_string()
+                };
+                Err(anyhow::anyhow!(message))
+            }
+        }
+    }
+
+    pub fn result_as_f64(&self) -> Result<f64> {
+        let x = self.result_as_u256()?;
+        let non_zero_bits = x.bit_len();
+        println!("Number of bits = {non_zero_bits}");
+        if non_zero_bits > f64::MANTISSA_DIGITS as usize {
+            println!("f64 is not big enough to accuratly represent integer with {non_zero_bits} non zero bits");
+        }
+        if non_zero_bits > u128::BITS as usize {
+            return Err(anyhow::anyhow!(
+                "u128 is not big enough to fit integer with {non_zero_bits} non zero bits"
+            ));
+        }
+        let be: [u8; 32] = x.to_be_bytes();
+        let (_a, b) = convert(be);
+        let u = u128::from_be_bytes(b);
+        Ok(u as f64)
+    }
+}
+
+impl RequestEthCall {
+    pub fn latest(calldata: &Bytes, contract_address: &Address, id: u64) -> RequestEthCall {
+        let params = RequestEthCallParams {
+            data: calldata.0.encode_hex_upper_with_prefix(),
+            from: "0x0000000000000000000000000000000000000000".to_string(),
+            to: contract_address.to_string(),
+        };
+
+        RequestEthCall {
+            jsonrpc: "2.0".to_string(),
+            method: "eth_call".to_string(),
+            id,
+            params: (params, "latest".to_string()),
+        }
+    }
+}
+
+async fn fetch_all(_resourses: &[FeedConfig]) -> Result<ResponseEthCall> {
+    //let rpc_url = "https://rpc.eth.gateway.fm".parse()?;
+    let rpc_url: Url = "https://eth.llamarpc.com".parse()?;
+    let yn_eth_x_contract_address = address!("0x657d9ABA1DBb59e53f9F3eCAA878447dCfC96dCb");
+
+    // Create a provider, in order to create instance of the contract
+    let provider = ProviderBuilder::new().on_http(rpc_url.clone());
+    let contact = YnETHx::new(yn_eth_x_contract_address, provider);
+    let shares = U256::from(1_000_000_000_000_000_000_u64);
+
+    let x = contact.convertToAssets(shares);
+    let calldata = x.calldata().clone();
+    let eth_call = RequestEthCall::latest(&calldata, &yn_eth_x_contract_address, 12345);
+    let request_body =
+        serde_json::to_string(&eth_call).expect("Serialization failed, this is impossible");
+
+    let mut req = Request::builder();
+    req.method(Method::Post);
+    req.header("Content-Type", "application/json");
+    req.uri(rpc_url.to_string());
+    req.body(request_body);
+
```

Comment:
Can we use this helper functions from the SDK here? - https://github.com/blocksense-network/blocksense/blob/main/libs/sdk/src/http.rs

Reply:
It's a POST request and the SDK provides only get request. I will extend the SDK

Reply:
done